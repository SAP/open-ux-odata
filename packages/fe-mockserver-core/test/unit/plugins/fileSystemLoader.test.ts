import FileSystemLoader from '../../../src/plugins/fileSystemLoader';
import * as path from 'path';

describe('Filesystem Loader', () => {
    const myFSLoader = new FileSystemLoader();
    it('can load json / text files', async () => {
        const jsonData = await myFSLoader.loadFile(path.join(__dirname, 'fixtures', 'myFile.json'));
        expect(jsonData).toMatchSnapshot();
    });
    it('can load js files', async () => {
        const jsonData = await myFSLoader.loadJS(path.join(__dirname, 'fixtures', 'myFile.js'));
        expect(jsonData.getData()).toMatchSnapshot();
    });
    it('can check file exists', async () => {
        let doesExist = await myFSLoader.exists(path.join(__dirname, 'fixtures', 'myFile.json'));
        expect(doesExist).toBe(true);
        doesExist = await myFSLoader.exists(path.join(__dirname, 'fixtures', 'notAtRealFile.json'));
        expect(doesExist).toBe(false);
    });
});
