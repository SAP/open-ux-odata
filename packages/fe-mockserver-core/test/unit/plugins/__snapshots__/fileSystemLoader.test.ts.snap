// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filesystem Loader can load js files 1`] = `
[
  {
    "name": "Something",
  },
]
`;

exports[`Filesystem Loader can load json / text files 1`] = `
"[
    {
        "name": "Value"
    },
    {
        "name": "Second one"
    }
]
"
`;
